ports/geckolib/wrapper.rs

```diff
@@ -95,16 +95,40 @@ impl<'ln> GeckoNode<'ln> {
     }
 }
 
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Debug)]
 pub struct GeckoRestyleDamage(nsChangeHint);
 
 impl TRestyleDamage for GeckoRestyleDamage {
     type PreExistingComputedValues = nsStyleContext;
+
+    fn empty() -> Self {
+        use std::mem;
+        GeckoRestyleDamage(unsafe { mem::transmute(0u32) })
+    }
+
     fn compute(source: Option<&nsStyleContext>,
                new_style: &Arc<ComputedValues>) -> Self {
         type Helpers = ArcHelpers<ServoComputedValues, ComputedValues>;
         let context = match source {
             Some(ctx) => ctx as *const nsStyleContext as *mut nsStyleContext,
+            // If there's no style source (that is, no style context), there can
+            // be two reasons for it.
+            //
+            // The first one, is that this is not an incremental restyle (so we
+            // also don't have the old computed values). In that case the best
+            // we can do is return rebuild_and_reflow.
+            //
+            // The second one is that this is an incremental restyle, but the
+            // node has display: none. In this case, the old computed values
+            // should still be present, and we should be able to compare the new
```

Comment:
Emphasize that they're stored on the node by servo, but we don't have a style context.